sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIOWriteExceptionHandlingTest.java

```diff
@@ -73,49 +70,43 @@ public static Collection<Object[]> data() {
     return Arrays.asList(
         new Object[][] {
           // DEADLINE_EXCEEDED is the only exception type that generates retries in-SDK
-          // The default backoff generates 9 retries, and then errors out the pipeline.
-          {ErrorCode.DEADLINE_EXCEEDED, "deadline passed!", 9, 10},
+          // The default backoff generates _ABOUT_ 9 retries, and then errors out the pipeline.
+          {ErrorCode.DEADLINE_EXCEEDED, "deadline passed!"},
```

Comment:
Rather than testing for DEADLINE_EXEEDED in the assert, put a hasRetries parameter here, and use that to determine whether to test no sleeps and 1 write, or writes >=9 and sleeps => 8